test/cell_test.rb

```diff
@@ -56,20 +56,20 @@ def test_fired_upon_starts_false
   end
 
   def test_fire_upon_lowers_ship_health_by_one
-    @cell.place_ship(@submarine)
+    @cell.place_ship(@cruiser)
     @cell.fire_upon
+
     assert_equal 1, @cell.ship.health
-    assert_equal 2, @cell.ship.health
   end
 
   def test_cell_has_not_been_fired_upon
-    assert_equal false, @cell.fired_upon?
-    @cell.fire_upon
-    assert_equal true, @cell.fired_upon?
+    @cell.place_ship(@cruiser)
+    @cell_2.place_ship(@submarine)
   end
 
   def test_fire_upon_reduces_health
     @cell.fire_upon
+    @cell_2.fire_upon
     assert_equal @ship.health, @ship.length - 1
```

Comment:
I think the reason this one didn't work (you said it returned nil) is because there isn't an @ship instance variable in this file; instead, we want to test the ship "held by" @cell and @cell_2, with something like this:
`assert_equal @cell.ship.health, @cell.ship.length - 1
 assert_equal @cell_2.ship.health, @cell_2.ship.length -1`